adoc/chapters/programming_interface.adoc

```diff
@@ -13229,10 +13229,16 @@ exposed in SYCL through [code]#+parallel_for (nd_range,...)+# and the
 grained control of the enqueuing of the kernel. This variation of
 parallel_for expects an [code]#nd_range#, specifying both local and
 global ranges, defining the global number of work-items and the number in
-each cooperating work-group. The resulting function object is passed an
-[code]#nd_item# instance making all the information available, as well
-as <<work-group-barrier>> to synchronize between the <<work-item>>s in
-the <<work-group>>.
+each cooperating work-group. The function object that represents the SYCL
+kernel function must take one of:
+1) a single SYCL [code]#nd_item# parameter, 2) a single generic parameter
+([code]#template# parameter or [code]#auto#), 3) any other type converted
```

Comment:
Should we state here someplace that the kernel argument's type will be `nd_item` if the formal parameter is declared `auto`?

Reply:
Added in 993559f.